### PR TITLE
Configure long line length 

### LINKED ISSUE #63639: Configure long lines length
Found no duplicated for this one

Currently it's hardcoded

https://github.com/Microsoft/vscode/blob/1ed635852b62e6c4cf3ea57bf465ee5528eeb720/src/vs/workbench/services/textMate/electron-browser/TMSyntax.ts#L452-L459

I propose a config entry for *long line* length

If you're object, I'd like to create pull request for this feature

Your thoughts?

### PR DESCRIPTION
**Pull request by ColCh**
closes #63639

### CODE REVIEW

File: src/vs/editor/common/config/commonEditorConfig.ts [hunk 1]
```diff
@@ -822,6 +822,11 @@ const editorConfiguration: IConfigurationNode = {
 			'default': EDITOR_DEFAULTS.contribInfo.lightbulbEnabled,
 			'description': nls.localize('codeActions', "Enables the code action lightbulb in the editor.")
 		},
+		'editor.longLineLength': {
+			'type': 'integer',
+			'default': EDITOR_DEFAULTS.contribInfo.longLineLength,
```
Comment (ColCh) [inline src/vs/editor/common/config/commonEditorConfig.ts]: is `contribInfo` a right place for this setting?

File: src/vs/workbench/services/textMate/electron-browser/TMSyntax.ts [hunk 2]
```diff
@@ -448,13 +463,13 @@ class TMTokenization implements ITokenizationSupport {
...
 		}
 
-		// Do not attempt to tokenize if a line has over 20k
-		if (line.length >= 20000) {
+		// Do not attempt to tokenize if a line is too long
+		if (line.length >= this._longLineLength) {
 			if (!this._tokenizationWarningAlreadyShown) {
 				this._tokenizationWarningAlreadyShown = true;
-				this.notificationService.warn(nls.localize('too many characters', "Tokenization is skipped for lines longer than 20k characters for performance reasons."));
+				this.notificationService.warn(nls.localize('too many characters', 'Tokenization is skipped for long lines for performance reasons.'));
```
Comment (alexr00) [inline src/vs/workbench/services/textMate/electron-browser/TMSyntax.ts]: Can you add a bit about how this can be configured now? "The length of a long line can be set by configuring editor.longLineLength in settings."

File: src/vs/editor/common/config/commonEditorConfig.ts (hunk 1 above)
Comment (alexr00) [reply src/vs/editor/common/config/commonEditorConfig.ts]: I think it would go better as EDITOR_DEFAULTS.longLineLength. @alexandrudima, can you confirm?

**Review by alexdima — CHANGES_REQUESTED** [review]
---

#### 1. Don't parse/validate the option in `editorOptions.ts` 

Editor options can be a bit tricky because each editor instance can be configured differently. e.g. it would be possible to have an editor with word wrap enabled while another one with word wrap not enabled. That is the reason why all these `editorOptions.ts` classes/interfaces exist. This is typically done by having someone on the outside reading the `IConfigurationService` and applying it partly or fully to an editor instance, and then all the code in the editor goes through these `editorOptions.ts` interfaces to read the effective setting.

However, in this case, this new option is not editor instance specific. This can be spotted by the reading side, in `TMSyntax.ts`, where reading is done through the `IConfigurationService` directly (a singleton). So, I suggest we remove all the additions from `editorOptions.ts`. It is still possible to have an option `editor.something` and for it to not be parsed/validated/etc. in `editorOptions.ts` if it is being read straight through the `IConfigurationService`.

---

#### 2. Find a better name

I think `editor.longLineLength` is too fuzzy. For example, we have another built-in constant for rendering lines. Lines longer than 10k chars will be painted with a '...' in the view (again to prevent freezes). If we were to make that configurable we would have a name clash. I therefore suggest we come up with a better name here, something that contains one of the following (or a synonym if you have a better one): tokenization, grammars, colorization, syntax highlighting.

**Review by alexr00 — CHANGES_REQUESTED** [review]
See comments

File: src/vs/editor/common/config/commonEditorConfig.ts (hunk 1 above)
Comment (ColCh) [reply src/vs/editor/common/config/commonEditorConfig.ts]: Renamed option name to `maxTokenizationLineLength`

**Comment by ColCh** [comment]
Thank you for such detailed review comments!

Updated the code, please take a look again @alexandrudima @alexr00 
Build seems to pass in CI

TL;DR; changes
* removed changes from **editorOptions.ts**. Didn't knew that, thanks.
* renamed option name to `maxTokenizationLineLength` to be more specific
* added note on how to change it

**Comment by alexandrudima** [comment]
:+1: Thank you!

**Comment by ColCh** [comment]
Thank you too! 👍

File: src/vs/editor/common/config/commonEditorConfig.ts (hunk 1 above)
Comment (lvleihere) [reply src/vs/editor/common/config/commonEditorConfig.ts]: god job!
